core/common/result.hpp

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_RESULT_HPP
+#define KAGOME_RESULT_HPP
+
+#include <ciso646>
+
+#include <boost/optional.hpp>
+#include <boost/variant.hpp>
+
+#include "common/visitor.hpp"
+
+/*
+ * Result is a type which represents value or an error, and values and errors
+ * are template parametrized. Working with value wrapped in result is done using
+ * match() function, which accepts 2 functions: for value and error cases. No
+ * accessor functions are provided.
+ */
+
+namespace kagome {
```

Comment:
We can write `namespace kagome::expected`. We can update other files as well

Reply:
C++17
+
+
1
7